### PR TITLE
IL-15906: Proper passing of parameters to packaged app

### PR DESCRIPTION
**Pull request by ericgibby**
This change adds code in the `onLaunched` event listener to record the URL used to launch the app via the URL handler. That URL is then stored in local storage so it can be retrieved along with server settings when the web view is loaded, passing the original parameters on to the WebGL app.

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -174,195 +176,214 @@ function backgroundAccept(sockInfo) {
...
+	//if (launchData.source == 'reload') { console.log('app was reloaded'); return }
+	if (launchData.source == 'restart') { console.log('chrome restarted'); return }
+
+	//console.log('onLaunched with launchdata',launchData)
+
+	// If launched via a url_handler, we need to grab data passed through the URL.
+	// We only want the data that comes after the base URL, specified by the ID
+	// of the url_handler.
+	var relativeUrl;
+	if (launchData.source === 'url_handler' && launchData.url && launchData.id) {
```
Comment (ericgibby) [inline background.js:209]: If the app was launched via a URL handler, that info will be passed in the event parameters. The naming convention used for the `id` properties in the `url_handlers` section of `manifest.json` matches the URL used to launch the app, allowing us to remove the unnecessary information from the beginning of the URL and leave just the portion used to pass parameters.

File: background.js [hunk 2]
```diff
@@ -174,195 +176,214 @@ function backgroundAccept(sockInfo) {
...
+		var regex = new RegExp('^http[s]?:\/\/' + launchData.id, 'i');
+		relativeUrl = launchData.url.replace(regex, '');
+	}
+
+	// If we identified a relative URL from the url_handler, we'll stuff
+	// it in local storage to be retrieved when the app launches.
+	if (relativeUrl) {
```
Comment (ericgibby) [inline background.js:216]: If the URL handler was used to launch the app, we'll store the URL parameters in local storage for retrieval later. If not, we'll clear it so it doesn't take the user somewhere unexpected.

File: browser.js [hunk 3]
```diff
@@ -3,218 +3,218 @@ window.onresize = doLayout;
...
 	console.log("got server settings");
 	console.log(serverSettings);
-	var url = convertServerSettingsToUrl(serverSettings);
-	
-	if (url && url.length > 0) {
-		console.log("calling navigateTo url: " + url);
-		navigateTo(url);
-	}
-	
+	chrome.storage.local.get('relativeLaunchUrl', function (obj) {
```
Comment (ericgibby) [inline browser.js:79]: Check to see if a launch URL was stored in local storage, so we can combine its parameters with the server settings stored in managed storage.

File: parseSettings.js [hunk 4]
```diff
@@ -1,37 +1,33 @@
...
-	if (serverSettings !== undefined && serverSettings !== null) {
-	
-		var protocol = 'http://';
-				
-		var domain = baseUrl;
-				
-		var queryString = '';
-		if (serverSettings && serverSettings.cloudSiteCode && serverSettings.cloudSiteCode.length > 0) {
-			queryString =  '?sitecode=' + serverSettings.cloudSiteCode;
+function convertServerSettingsToUrl(serverSettings, launchUrl) {
```
Comment (ericgibby) [inline parseSettings.js:4]: This now includes the original launch URL (if there was one), so those parameters can be passed on to the WebGL app along with the server settings.

File: manifest.json [hunk 5]
```diff
@@ -43,11 +43,41 @@
         "128": "images/il_128.png"
     },
     "url_handlers": {
-        "openApp": {
+        "localhost/player": {
```
Comment (ericgibby) [inline manifest.json:46]: The various IDs used make it easier to extract the pertinent information from the URL. It allows us to build a simple regex to remove the beginning of the URL and leave the relevant path and query string info in place.

**Review by adamjchristiansen — APPROVED** [review]

**Pull request merged**